### DISCUSSION TITLE
Collaboration between Vike and Solid

### DISCUSSION THREAD

**Discussion by brillout** [solidjs/solid#1845] [General]
Author here of [vite-plugin-ssr](https://vite-plugin-ssr.com/) and Vike (the upcoming rebrand of vite-plugin-ssr).

Some updates on Vike and why I believe it would make sense for us to collaborate in a deep fashion.

## Long-term vision

Beyond technical considerations, there is one aspect I'd like to emphasize: the long-term vision of Vike which is to become an open foundation to build upon.

Everything about Vike has been designed around extensibility and lower-level primitives. For example, Vike is the only framework that can deeply integrate with Relay. The long-term goal being to be able to build all kinds of frameworks such as Gatsby's data layer or Astro's island architecture.

There is an example of a Solid integration by @magne4000: [vike-solid](https://github.com/magne4000/vike-solid). While some features are missing, it shows that Vike can take over most of the framework work while providing enough flexibility for the Solid team to innovate.

## No vendor lock in

We believe the fact that Vercel, Next.js, and React are getting increasingly tightly coupled with each other is a terrible outcome for users.

Users want freedom, and rightfully so: being vender-locked is a major threat for companies. Most companies are aware of that.

Vike unlocks a new kind of freedom: users can choose React, Solid, Vue, or whatever UI framework they want. (Astro as well but its architecture is too rigid, e.g. I don't see RSC being implementable with Astro.)

Telling users "pick Solid because of its merits" is an appealing story.

## Being able to use Solid while tapping into React's ecosystem

A major USP is that users can use both Solid and React for the same app.

Solid can be used for most pages, while React can be used only for a couple of pages that need libraries that only exists for React (that's actually a use case @samuelstroschein is using).

This also means that users can more seamlessly migrate between React and Solid.

The story being: "Try Solid and if you don't like it you can fallback to React or Vue. Although we believe you'll end up liking Solid a lot!"

## Built-in Router

Thanks to Vike's built-in rotuer, Vike brings a *lot* more features to the table than alternatives that are router agnostic.

The goal of the [V1 release](https://github.com/brillout/vite-plugin-ssr/issues?q=is%3Aissue+is%3Aopen+label%3A%22v1+release+%3Astar2%3A%22) is to be feature complete. The router will include all features users may expect: type safe routes, route guards, URL rewriting, being able to define all routes in a Single Route File, ...

All the Solid team has to do is to implement the Solid integration. Enabling Solid's team to focus on what makes Solid special.

## Vike roadmap

 - [V1 release](https://github.com/brillout/vite-plugin-ssr/issues?q=is%3Aissue+is%3Aopen+label%3A%22v1+release+%3Astar2%3A%22)
 - Vike rebrand. With a marketing that targets a wider audience with a more professional/dry tone. (The vite-plugin-ssr copy has more of a personal/hobbyist/passionate tone.)
 - Extend and polish [vike-solid](https://github.com/magne4000/vike-solid), [vike-react](https://github.com/vikejs/vike-react), and vike-vue.
 - Grow from 14k weekly downloads to 100k. (To put it in perspective: Solid Start is 8k weekly downloads, Astro went from 24k last year to now 114k, Remix went from 54k last year to now 174k.)

Once the V1 is released, Vike can be considered feature complete, highly polished, and robust. AFAICT the V1 alone is enough to grow to 100k, but we'll also invest in marketing to grow beyond 100k.

From a user yesterday:

 > I appreciate the descriptive and helpful error messages of vite-plugin-ssr more than once.

That's just the tip of the iceberg of the amount of meticulous care put into details. The V1 release will be the pinnacle of that.

## Vike visions

Vike is only the beginning. Some projects we're also working on:
 - [Bati](https://batijs.github.io/) (@magne4000 @aurelienlourot)
 - [Stem](https://stemjs.com/)
 - Developing novel open source business models (paramount for long-term health and gaining trust from users)

We believe Bati and Stem to be the future of frameworks, eventually bringing a 10x improvement over Next.js & co. Although it's going to take a while until we get there as we first focus on the getting the foundation right.

## Proposal

@davedbase @nksaraf @ryansolid I suggest the following:
 1. We discuss the potential issues you see.
 2. We implement a POC, e.g. extending [vike-solid](https://github.com/magne4000/vike-solid) to include island hydration and/or Server Components.

Ultimately, the best way to know whether a deep collaboration between the Vike and Solid team works out is to try it out. Personally, I'm very much looking forward to give it a shot.

**Comment by samuelstroschein** [comment]
_The remarks beneath are expressed by believing that Vike fulfills Solid Start's requirements based on our own production usage of Solid + Vike._

**Thoughts from us (inlang), who use Solid and Vike in production, on SolidStart building on Vike:**

- SolidStart can (likely?) be released in a few weeks. Our production Vike plus Solid setup is [maybe 300 lines of code](https://github.com/inlang/inlang/tree/main/source-code/website/src/renderer).
- SolidStart's appeal towards enterprises is increased by not having a lock-in towards Solid. Worst case, a migration to React/Vue/Angular is possible without ripping out the meta-framework. 
- SolidStart will likely be better when Vike is used under the hood because a) Vike is production ready, b) Vike will receive more resources because Vike appeals to a larger audience (Solid + React + Angular + Vue ...)

Besides the assumption that SolidStart benefits from building on Vike, we are likely able to justify a sponsorship of SolidStart x Vike for them to a) make the integration work and then b) deliver an i18n solution based on inlang's core that works in both Vike and SolidStart. The latter i18n solution needs to be maintained, entailing a recurring and long-running sponsorship.

**Reply by brillout** [reply]
Very much looking forward to repeat the success of Inlang's SvelteKit SDK with the Vike + Solid (Start) combo.

Inlang's (increased) sponsoring would also be very much appreciated.
